src/rrd2json.c

```diff
@@ -176,7 +176,7 @@ static inline size_t prometheus_name_copy(char *d, const char *s, size_t usable)
     for(n = 0; *s && n < usable ; d++, s++, n++) {
         register char c = *s;
 
-        if(unlikely(!isalnum(c))) *d = '_';
+        if(unlikely(c != '.' && !isalnum(c))) *d = '_';
```

Comment:
No, this isn't the right logic.

When copying the hostname, instead of calling `prometheus_name_copy`, you should call a new function which just copies character-by-character, except for three special cases of double-quote, backslash and newline (\n), in which case it precedes the character with a backslash. [Specification](https://prometheus.io/docs/instrumenting/exposition_formats/#text-format-details):

> label_value can be any sequence of UTF-8 characters, but the backslash, the double-quote, and the line-feed characters have to be escaped as \\, \", and \n, respectively

When copying the chart/dimension, then continue to call `prometheus_name_copy` and this *should* continue to replace dot with underscore.  [Specification](https://prometheus.io/docs/practices/naming/):

> The metric name ... may contain ASCII letters and digits, as well as underscores and colons. It must match the regex `[a-zA-Z_:][a-zA-Z0-9_:]*`.